omnipaxos/src/sequence_paxos/leader.rs

```diff
@@ -219,43 +219,52 @@ where
             .leader_state
             .get_decided_idx(*pid)
             .expect("Received PromiseMetaData but not found in ld");
-        let (delta_snapshot, suffix, sync_idx) =
+        let (delta_snapshot, suffix, sync_idx, compacted_idx) =
             if (promise_n == max_promise_n) && (promise_accepted_idx < max_accepted_idx) {
                 if self.internal_storage.get_compacted_idx() > *promise_accepted_idx
                     && T::Snapshot::use_snapshots()
                 {
                     let delta_snapshot = self
                         .internal_storage
-                        .create_diff_snapshot(follower_decided_idx, my_decided_idx)
+                        .create_diff_snapshot(follower_decided_idx)
                         .expect("storage error while trying to read diff snapshot");
                     let suffix = self
                         .internal_storage
                         .get_suffix(my_decided_idx)
                         .expect("storage error while trying to read log suffix");
-                    (Some(delta_snapshot), suffix, follower_decided_idx)
+                    let compacted_idx = match self.internal_storage.stopsign_is_decided() {
+                        true => my_decided_idx - 1,
+                        false => my_decided_idx,
+                    };
```

Comment:
The problem is we are creating a snapshot diff for the follower and the leader doesn't adopt this diff themself. Because the leader doesn't adopt the snapshot we can't use the leader's `get_compacted_idx()`,  and instead calculate it (snapshot always goes up to leaders decided (-1 if decided stopsign)).

Reply:
If the leader would adopt the snapshot here it would have to create the snashot diff (follower_dec, leader_dec) for the follower, then create and adopt a snapshot (leader_compactIdx, leader_dec), and then we could rely on the leader's `get_compacted_idx()`.

Reply:
Right, that's true.